src/state/tokenPrices/hooks.ts

```diff
@@ -0,0 +1,70 @@
+import { ChainId } from '@kyberswap/ks-sdk-core'
+import { useEffect, useMemo } from 'react'
+
+import { NETWORKS_INFO } from 'constants/networks'
+import { useActiveWeb3React } from 'hooks'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { isAddressString } from 'utils'
+
+import { updatePrices } from '.'
+
+export const useTokenPrices = (addresses: Array<string>) => {
+  const tokenPrices = useAppSelector(state => state.tokenPrices)
+  const dispatch = useAppDispatch()
+  const { chainId } = useActiveWeb3React()
+
+  const addressKeys = addresses
+    .sort()
+    .map(x => x.toLowerCase())
+    .join(',')
+
+  const tokenList = useMemo(() => {
+    return addressKeys.split(',').filter(Boolean)
+  }, [addressKeys])
+
+  const unknownPriceList = useMemo(() => {
+    return tokenList.filter(item => tokenPrices[`${item}_${chainId}`] === undefined)
+  }, [tokenList, chainId, tokenPrices])
+
+  useEffect(() => {
+    const fetchPrices = async () => {
+      const chainIdMapping: Partial<Record<ChainId, string>> = {
+        [ChainId.BSCMAINNET]: 'bsc',
+        [ChainId.BTTC]: 'bttc',
+      }
+      const chainString = chainId ? chainIdMapping[chainId] || NETWORKS_INFO[chainId].route : ''
+      const res = await fetch(
+        `${process.env.REACT_APP_PRICE_API}/${chainString}/api/v1/prices/?ids=${unknownPriceList.join(',')}`,
+      ).then(res => res.json())
+
+      if (res?.data?.prices?.length) {
+        const formatedPrices = unknownPriceList.map(address => {
+          const price = res.data.prices.find(
+            (p: { address: string; marketPrice: number; price: number }) => p.address.toLowerCase() === address,
+          )
+
+          return {
+            address,
+            chainId: chainId || ChainId.MAINNET,
+            price: price?.marketPrice || price?.price || 0,
+          }
+        })
+
+        dispatch(updatePrices(formatedPrices))
```

Comment:
![image](https://user-images.githubusercontent.com/23650671/194992641-4263a723-5fa2-4765-9109-c448726f8e50.png)
```suggestion
        dispatch(updatePrices(formattedPrices))
```